pkg/vegeta/report.go

```diff
@@ -19,11 +21,50 @@ const (
 	// TextFormat typedef for query param "text"
 	TextFormat Format = "text"
 	// HistogramFormat typedef for query param "histogram"
-	//HistogramFormat Format = "histogram"
+	HistogramFormat Format = "histogram"
 	// BinaryFormat typedef for query param "binary"
 	BinaryFormat Format = "binary"
+	// DefaultBucketString Default Bucket String
+	DefaultBucketString string = "0,500ms,1s,1.5s,2s,2.5s,3s"
 )
 
+// SplitFormat splits the Format into sub-Formats separated by '-' using StringsToFormat
+func (f Format) SplitFormat() (fs []Format) {
+	sf := string(f)
+	sli := strings.Split(sf, "-")
+	for _, s := range sli {
+		fs = append(fs, Format(s))
+	}
+	return
+}
+
+// GetFormat returns the actual Format string. It is useful in case of Clubbed Format string
+func (f Format) GetFormat() Format {
+	fs := f.SplitFormat()
+	return fs[0]
+}
+
+// GetTimeBucketOfHistogram returns time duration Bucket for the histogram format string
+func (f Format) GetTimeBucketOfHistogram() (buk []byte, err error) {
+	fs := f.SplitFormat()
+	if fs[0] != HistogramFormat {
+		return buk, errors.New("time bucket can be obtained by histogram format only")
```

Comment:
Let stick to `fmt.Errorf()` instead of introducing `errors.New()`. I think `errors.New` is pretty much deprecated now.

Reply:
I don't think. It's just that `fmt.Errorf()` is more convenient in handling format strings, though benchmarks are in favor of `errors.New()` for simple string messages. https://play.golang.org/p/gXzQC2Ux4J

Though it doesn't matter, will change it for the consistency.